drivers/bluetooth/hci/h5.c

```diff
@@ -768,10 +768,29 @@ static int h5_open(void)
 	return 0;
 }
 
+static void h5_deinit(void)
+{
+	k_work_cancel_delayable(&retx_work);
+	k_work_cancel_delayable(&ack_work);
+
+	k_thread_abort(&rx_thread_data);
+	k_thread_abort(&tx_thread_data);
+}
+
+static int h5_close(void)
+{
+	h5_deinit();
+
+	uart_irq_rx_disable(h5_dev);
```

Comment:
disable the interrupt first to be sure that we can cancel the works and kill the rx thread.